staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -275,76 +272,20 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	crdName := requestInfo.Resource + "." + requestInfo.APIGroup
-	// HACK: support the case when we add core resources through CRDs (KCP scenario)
-	if requestInfo.APIGroup == "" {
-		crdName = crdName + "core"
-	}
-
-	var crdClusterName string
-	var crd *apiextensionsv1.CustomResourceDefinition
-	var err error
-	handleErrorFunc := func(err error) {
-		if apierrors.IsNotFound(err) {
-			if !r.hasSynced() {
-				responsewriters.ErrorNegotiated(serverStartingError(), Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req)
-				return
-			}
-			r.delegate.ServeHTTP(w, req)
-		}
-		if err != nil {
-			utilruntime.HandleError(err)
-			responsewriters.ErrorNegotiated(
-				apierrors.NewInternalError(fmt.Errorf("error resolving resource: %v", err)),
-				Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
-			)
-		}
+	crd, err := r.crdLister.GetWithContext(ctx, crdName)
+	if apierrors.IsNotFound(err) {
+		r.delegate.ServeHTTP(w, req)
+		return
 	}
-
-	cluster, err := genericapirequest.ValidClusterFrom(ctx)
 	if err != nil {
+		utilruntime.HandleError(err)
 		responsewriters.ErrorNegotiated(
-			apierrors.NewInternalError(fmt.Errorf("error resolving resource: %v", err)),
+			apierrors.NewInternalError(fmt.Errorf("error resolving resource")),
 			Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
 		)
 		return
 	}
 
-	if cluster.Wildcard {
```

Comment:
How does discovery for `*` work now?

Reply:
This code was moved to kcp - no net changes.